example/spec/parallel/ingest/ingestFromPdrWithChildWorkflowMetaSpec.js

```diff
@@ -0,0 +1,181 @@
+ * performs the fake processing step - generates CMR metadata
+ * Moves the file to the final location
+ * Does not post to CMR (that is in a separate test)
+ */
+
+const { randomString } = require('@cumulus/common/test-utils');
+const { deleteS3Object } = require('@cumulus/aws-client/S3');
+const { s3 } = require('@cumulus/aws-client/services');
+const { LambdaStep } = require('@cumulus/integration-tests/sfnStep');
+
+const {
+  addCollections,
+  addProviders,
+  api: apiTestUtils,
+  buildAndExecuteWorkflow,
+  cleanupProviders,
+  cleanupCollections,
+  getExecutionInputObject,
+  waitForStartedExecution,
+} = require('@cumulus/integration-tests');
+
+const {
+  createTestDataPath,
+  createTestSuffix,
+  createTimestampedTestId,
+  deleteFolder,
+  loadConfig,
+  uploadTestDataToBucket,
+  updateAndUploadTestDataToBucket,
+} = require('../../helpers/testUtils');
+
+const lambdaStep = new LambdaStep();
+const workflowName = 'DiscoverAndQueuePdrsChildWorkflowMeta';
+const origPdrFilename = 'MOD09GQ_1granule_v3.PDR';
+
+const s3data = [
+  '@cumulus/test-data/pdrs/MOD09GQ_1granule_v3.PDR',
+];
+
+const unmodifiedS3Data = [
+  '@cumulus/test-data/granules/MOD09GQ.A2016358.h13v04.006.2016360104606.hdf.met',
+  '@cumulus/test-data/granules/MOD09GQ.A2016358.h13v04.006.2016360104606.hdf',
+];
+
+describe('The DiscoverAndQueuePdrsChildWorkflowMeta workflow', () => {
+  const providersDir = './data/providers/s3/';
+  const collectionsDir = './data/collections/s3_MOD09GQ_006';
+
+  let beforeAllFailed;
+  let config;
+  let pdrFilename;
+  let provider;
+  let testDataFolder;
+  let testSuffix;
+  let workflowExecution;
+  let addedCollection;
+  let executionNamePrefix;
+  let queuePdrsOutput;
+
+  beforeAll(async () => {
+    try {
+      config = await loadConfig();
+
+      process.env.PdrsTable = `${config.stackName}-PdrsTable`;
+
+      const testId = createTimestampedTestId(config.stackName, 'IngestFromPdr');
+      testSuffix = createTestSuffix(testId);
+      testDataFolder = createTestDataPath(testId);
+
+      pdrFilename = `${testSuffix.slice(1)}_${origPdrFilename}`;
+
+      provider = { id: `s3_provider${testSuffix}` };
+
+      // populate collections, providers and test data
+      const populatePromises = await Promise.all([
+        updateAndUploadTestDataToBucket(
+          config.bucket,
+          s3data,
+          testDataFolder,
+          [
+            { old: 'cumulus-test-data/pdrs', new: testDataFolder },
+            { old: 'DATA_TYPE = MOD09GQ;', new: `DATA_TYPE = MOD09GQ${testSuffix};` },
+          ]
+        ),
+        uploadTestDataToBucket(
+          config.bucket,
+          unmodifiedS3Data,
+          testDataFolder
+        ),
+        addCollections(config.stackName, config.bucket, collectionsDir, testSuffix, testId),
+        addProviders(config.stackName, config.bucket, providersDir, config.bucket, testSuffix),
+      ]);
+
+      addedCollection = populatePromises[2][0];
```

Comment:
Super small nit but maybe we could destructure the return from `Promise.all` and switch up the order so that we don't have to index into it here. Something like:

```
const [addedCollection, ...rest] = await Promise.all([
        addCollections(config.stackName, config.bucket, collectionsDir, testSuffix, testId),
        updateAndUploadTestDataToBucket(
          config.bucket,
          s3data,
          testDataFolder,
          [
            { old: 'cumulus-test-data/pdrs', new: testDataFolder },
            { old: 'DATA_TYPE = MOD09GQ;', new: `DATA_TYPE = MOD09GQ${testSuffix};` },
          ]
        ),
        uploadTestDataToBucket(
          config.bucket,
          unmodifiedS3Data,
          testDataFolder
        ),
        addProviders(config.stackName, config.bucket, providersDir, config.bucket, testSuffix),
      ]);
```

Reply:
https://github.com/nasa/cumulus/pull/2114/commits/b2951e1aeffe4f886a6a361104dbc8c01edc988f